tl.tl

```diff
@@ -4897,11 +4935,12 @@ tl.type_check = function(ast: Node, opts: TypeCheckOptions): {Error}, {Error}, T
    local lax = opts.lax
    local filename = opts.filename
 
-   local result = opts.result or {
-      syntax_errors = {},
-      type_errors = {},
-      unknowns = {},
-      warnings = {},
+   local result = {
+      syntax_errors = opts.result and opts.result.syntax_errors or {},
+      type_errors = opts.result and opts.result.type_errors or {},
+      unknowns = opts.result and opts.result.unknowns or {},
+      warnings = opts.result and opts.result.warnings or {},
+      dependencies = opts.result and opts.result.dependencies or {},
```

Comment:
I have passed around these lists from `result` to `result` in order to combine errors from multiple files into a single list. If you want each file's `result` to have only its list of direct dependencies, I think each `result` needs a fresh `dependencies` table.